agent/src/util/device_plugin_service.rs

```diff
@@ -247,10 +295,27 @@ impl InstanceDevicePlugin {
                 dps.instance_name
             );
 
-            let virtual_devices =
-                build_list_and_watch_response(dps.clone(), kube_interface.clone())
-                    .await
-                    .unwrap();
+            let device_usage_state = get_instance_device_usage_state(
+                &dps.node_name,
+                &dps.instance_name,
+                &dps.config_namespace,
+                &dps.config.capacity,
+                kube_interface.clone(),
+            )
+            .await;
+
+            let virtual_devices = device_usage_state
+                .into_iter()
+                .map(|(id, state)| v1beta1::Device {
+                    id,
+                    health: match state {
+                        DeviceUsageStatus::Free | DeviceUsageStatus::ReservedByInstance => {
+                            HEALTHY.to_string()
```

Comment:
might be helpful to add some comments here, why Free and ReservedByInstance are HEALTHY, but ReservedByConfiguration is considered UNHEALTHY

Reply:
i think this might be difficult to maintain without some of the documentation (https://github.com/project-akri/akri-docs/blob/725ec5e202062787bda2257ee2260dc12442970a/proposals/configuration-level-resources.md) being in the code.

Reply:
that's correct.  Before adding CL resource support, a slot can either be 
free,
allocated by Instance dp on this node or 
allocated by other nodes.

for Instance dp, a slot is considerred healthy if it's free or allocated by Instance dp on this node

After CL resource support, there are two types of device plugin (Instance and Configuration dp) running on a node. A slot can be 
free, 
allocated by Instance dp on this node
allocated by Configuration dp on this node
allocated by other nodes

for Instance dp, a slot is considered healthy if it's free or allocated by Instance dp on this node
for Configuration dp, a slot is considered healthy if it's free or allocated by Configuration dp on this node

Reply:
i think a comment for list_and_watch with that info would be quite useful

Reply:
add comment